### PR TITLE
Add support for python3

### LINKED ISSUE #502: Support for Python3
### Description
Please leave a brief discription of the bug or feature request: Python 2.7 is depreciated , so we can add support for python version 3 ,as scancode-toolkit is also updated to that version.
* if reporting a bug, tell us how to reproduce the issue. `run python build.py`
* Check the Develop Console tab and copy any error text there might be: https://scancode-workbench.readthedocs.io/en/develop/basics/check_for_errors_in_the_developer_tools.html

### System configuration
For bug reports, it really helps us to know:
* What OS are you running on? (Windows)
* What version of scancode-workbench are you running on? Latest

### PR DESCRIPTION
**Pull request by OmkarPh**
Fixes #502

### CODE REVIEW

**Comment by OmkarPh** [comment]
I had to modify build.py to fix the err occurring when running `python3 build.py`. It's fixed for good.
python2 & 3 both work fine with these changes
image

**Comment by OmkarPh** [comment]
@pombredanne ^^

**Comment by OmkarPh** [comment]
Also, I realised just now that the `^` before versions may install latest versions, hence removed them

File: .travis.yml [hunk 1]
```diff
@@ -4,8 +4,8 @@ matrix:
       dist: xenial
       sudo: required
       language: node_js
-      node_js: 10
-      python: 2.7.13
+      node_js: 16
+      python: 3.10.1
```
Comment (pombredanne) [inline .travis.yml]: Let's build with Python 3.7 or 3.8. #.10 is really bleeding edge and too recent IMHO.

File: .travis.yml [hunk 2]
```diff
@@ -32,7 +32,7 @@ matrix:
 
     - os: linux
       dist: xenial
```
Comment (pombredanne) [inline .travis.yml]: We likely want to use a more modern image. Xenial is Ubuntu 16, which is 6 years old.
Using 18 would be fine

**Review by pombredanne — CHANGES_REQUESTED** [review]
Thanks!
test-wise we likely want to upgrade de linux version and use a not such an advanced Python version... 3.or may be 3.7? being conservative there is best IMHO.

File: .travis.yml (hunk 1 above)
Comment (OmkarPh) [reply .travis.yml]: ok, is 3.7 good?

File: .travis.yml (hunk 2 above)
Comment (OmkarPh) [reply .travis.yml]: Ok 👍

Comment (OmkarPh) [reply .travis.yml]: hi, why is the block for linux repeated in travis.yml file 3 times. Shouldn't it be just once?

File: .travis.yml (hunk 1 above)
Comment (pombredanne) [reply .travis.yml]: yes :+1:

File: .travis.yml (hunk 2 above)
Comment (pombredanne) [reply .travis.yml]: Good catch!
we were testing on different Python versions which is not super interesting! we shoud instead test on different linux version.

File: build.py [hunk 3]
```diff
@@ -134,17 +134,17 @@ def get_git_version():
     # this may fail with exceptions
     cmd = 'git', 'describe', '--tags', '--long', '--dirty',
     version = subprocess.check_output(cmd, stderr=subprocess.STDOUT).strip()
```
Comment (pombredanne) [inline build.py:136]: You probably would rather decode early using this: then the `b` prefix is not needed afterwards
```suggestion
    version = subprocess.check_output(cmd, encoding='utf-8', stderr=subprocess.STDOUT).strip()
```

**Review by pombredanne — CHANGES_REQUESTED** [review]
I just have a minor nit picking for your review... This is looking good otherwise!
:+1:

File: build.py (hunk 3 above)
Comment (OmkarPh) [reply build.py:136]: Hi, I tried that, but it gives error unexpected keyword arg 'encoding' on pyhon2
It works fine with python3 latest versions, but not with python 2.
Also, this guy faced a same issue: https://github.com/eliben/pycparser/issues/296#issuecomment-444810928

I've done decoding on the next line though, thereby removing need for b prefix

**Comment by OmkarPh** [comment]
did the decode for npm_bin too
This looks much cleaner now ✌️

File: build.py (hunk 3 above)
Comment (pombredanne) [reply build.py:136]: Fair enough... we want to drop Python2 support alright!

**Review by pombredanne — APPROVED** [review]
LGTM!
Thank you ++

**Comment by pombredanne** [comment]
Thanks again!

**Pull request merged**